ProcessMaker/processmaker
# Feature/FOUR-20298: Implement logic to detect previous Customized Home Screen and redirect new Home Screen

## How to Test

Scenario 1: Without Package-dynamic-ui installed
- Login PM4
- First Page loaded after Login should be TASKS

Scenario 2: With package-dynamic-ui Installed

Use Case 1: Package Dynamic UI is installed and Default Welcome Screen is enabled by default
- Login PM4
- First Page loaded after login should be TASKS instead of DEFAULT WELCOME SCREEN

Use Case 2: Package Dynamic UI is installed and we need to create a Custom Dashboard  for Default Screen
- Login PM4
- Go to Screens -> Create Display Screen called "CUSTOM DASHBOARD TEST" 
- Add some controls to Display Screen and SAVE
- Go to Admin -> Customize UI -> Dashboards TAB -> Create new Dashboard and select screen "CUSTOM DASHBOARD TEST"
- Go to Admin -> Select some user (like Admin) and EDIT
- In Dynamic UI section, choose "My Dashboard" in Home Page dropdown
- Then select "CUSTOM DASHBOARD TEST" in Dashboard Dropdown
- Save changes.
- Login PM4 with Admin user
- First Page Loaded after login should be "CUSTOM DASHBOARD TEST"

## Related Tickets & Packages
https://processmaker.atlassian.net/browse/FOUR-20298

## Code Review Checklist
